### PR TITLE
interfaces/interfaces/scsi_generic: add interface for scsi generic de…

### PR DESCRIPTION
**Pull request by kubiko**
add interface for scsi generic devices

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: interfaces/builtin/scsi_generic.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+const scsiGenericConnectedPlugAppArmor = `
+# allow read,write access to generic scsi devices
+# ref: https://www.kernel.org/doc/Documentation/scsi/scsi-generic.txt
+/dev/sg[0-9]* rw,
+`
+
+var scsiGenericConnectedPlugUDev = []string{
+	 // ref: https://www.kernel.org/doc/Documentation/scsi/scsi-generic.txt
+	`KERNEL=="sg[0-9]*"`,
```
Comment (anonymouse64) [inline interfaces/builtin/scsi_generic.go:40]: I think this rule is fine

File: interfaces/builtin/scsi_generic.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const scsiGenericSummary = `allows access to generic driver (SCSI)devices`
```
Comment (pedronis) [inline interfaces/builtin/scsi_generic.go:22]: this looks odd, maybe instead:

`allows access to SCSI generic driver devices`

File: interfaces/builtin/scsi_generic.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+const scsiGenericBaseDeclarationSlots = `
+  scsi-generic:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
```
Comment (pedronis) [inline interfaces/builtin/scsi_generic.go:30]: shouldn't it be as constrained on the plug side as block-devices?

File: interfaces/builtin/scsi_generic.go (hunk 2 above)
Comment (kubiko) [reply interfaces/builtin/scsi_generic.go:22]: Updated as proposed

File: interfaces/builtin/scsi_generic.go (hunk 3 above)
Comment (pedronis) [reply interfaces/builtin/scsi_generic.go:30]: but as we know from the use case there's can be almost anything behind these devices, while optical-drive are assumed external. So there can be much larger effects writing  to these. So I think we need to be cautious and assume it's more like block-devices than optical-drive.

Comment (kubiko) [reply interfaces/builtin/scsi_generic.go:30]: Plug restriction has been now added.

Comment (anonymouse64) [reply interfaces/builtin/scsi_generic.go:30]: looks correct to me

**Review by anonymouse64 — APPROVED** [review]
lgtm, but I'm not sure if we won't need to modify some tests for the base declaration rules since this is a new super-privileged interface

**Review by mardy — APPROVED** [review]
LGTM!

**Review by pedronis — APPROVED** [review]
lgtm, still needs security review

File: interfaces/builtin/scsi_generic_test.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
```
Comment (alexmurray) [inline interfaces/builtin/scsi_generic_test.go:4]: Should the Copyright date be 2021?

**Review by alexmurray — APPROVED** [review]
LGTM!

File: interfaces/builtin/scsi_generic_test.go (hunk 4 above)
Comment (anonymouse64) [reply interfaces/builtin/scsi_generic_test.go:4]:
```suggestion
 * Copyright (C) 2021 Canonical Ltd
```

File: interfaces/builtin/scsi_generic.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
```
Comment (anonymouse64) [inline interfaces/builtin/scsi_generic.go:4]:
```suggestion
 * Copyright (C) 2021 Canonical Ltd
```

**Pull request merged**